consensus/src/main/kotlin/maru/consensus/qbft/QbftProtocolFactory.kt

```diff
@@ -0,0 +1,271 @@
+    val proposerSelector = ProposerSelectorImpl
+    val besuValidatorProvider = QbftValidatorProviderAdapter(validatorProvider)
+    val sealedBeaconBlockImporter =
+      createSealedBeaconBlockImporter(
+        executionLayerManager = executionLayerManager,
+        beaconChain = beaconChain,
+        proposerSelector = proposerSelector,
+        localNodeIdentity = localValidator,
+        stateTransition = stateTransition,
+      )
+
+    val qbftBlockCreatorFactory =
+      QbftBlockCreatorFactory(
+        manager = executionLayerManager,
+        proposerSelector = qbftProposerSelector,
+        validatorProvider = validatorProvider,
+        beaconChain = beaconChain,
+        finalizationStateProvider = finalizationStateProvider,
+        blockBuilderIdentity = Validator(localAddress.toArray()),
+        eagerQbftBlockCreatorConfig =
+          EagerQbftBlockCreator.Config(
+            maruConfig.qbftOptions.communicationMargin,
+          ),
+        nextBlockTimestampProvider = nextBlockTimestampProvider,
+        clock = clock,
+      )
+
+    val besuForksSchedule = ForksScheduleAdapter(forkSpec, maruConfig.qbftOptions)
+
+    val bftExecutors = BftExecutors.create(metricsSystem, BftExecutors.ConsensusType.QBFT)
+    val bftEventQueue = BftEventQueue(maruConfig.qbftOptions.messageQueueLimit)
+    val roundTimer =
+      RoundTimer(
+        /* queue = */ bftEventQueue,
+        /* baseExpiryPeriod = */ maruConfig.qbftOptions.roundExpiry.toJavaDuration(),
+        /* bftExecutors = */ bftExecutors,
+      )
+    val blockTimer = BlockTimer(bftEventQueue, besuForksSchedule, bftExecutors, clock)
+    val finalState =
+      QbftFinalStateAdapter(
+        localAddress = localAddress,
+        nodeKey = nodeKey,
+        validatorProvider = besuValidatorProvider,
+        proposerSelector = qbftProposerSelector,
+        validatorMulticaster = validatorMulticaster,
+        roundTimer = roundTimer,
+        blockTimer = blockTimer,
+        blockCreatorFactory = qbftBlockCreatorFactory,
+        clock = clock,
+        beaconChain = beaconChain,
+      )
+
+    val minedBlockObservers = Subscribers.create<QbftMinedBlockObserver>()
+    minedBlockObservers.subscribe { qbftBlock ->
+      newBlockHandler.handleNewBlock(qbftBlock.toBeaconBlock())
+      bftEventQueue.add(QbftNewChainHead(qbftBlock.header))
+    }
+
+    val blockImporter = QbftBlockImporterAdapter(sealedBeaconBlockImporter)
+
+    val blockCodec = QbftBlockCodecAdapter()
+    val blockInterface = QbftBlockInterfaceAdapter()
+    val protocolSchedule =
+      QbftProtocolScheduleAdapter(
+        blockImporter = blockImporter,
+        beaconChain = beaconChain,
+        proposerSelector = proposerSelector,
+        stateTransition = stateTransition,
+        executionLayerManager = executionLayerManager,
+      )
+    val messageValidatorFactory =
+      MessageValidatorFactory(
+        /* proposerSelector = */ qbftProposerSelector,
+        // protocolSchedule =
```

Comment:
nit: comment formatting is broken for `protocolSchedule` and `validatorProvider` arguments